VersionControlUI/src/nodes/DiffComparisonPair.h

```diff
@@ -61,27 +63,37 @@ class VERSIONCONTROLUI_API DiffComparisonPair : public Super<Visualization::UINo
 			Model::Text* newVersionObjectPath();
 			Model::Text* oldVersionObjectPath();
 
+			Model::Text* componentType();
+
 
 		private:
 			Model::Node* oldVersionNode_;
 			Model::Node* newVersionNode_;
+
 			Model::Text* newVersionObjectPath_;
 			Model::Text* oldVersionObjectPath_;
 
+			Model::Text* componentType_;
+
 
 };
 
 inline void DiffComparisonPair::setOldVersionNode(Model::Node* oldVersionNode) {oldVersionNode_ = oldVersionNode;}
 inline void DiffComparisonPair::setNewVersionNode(Model::Node* newVersionNode) {newVersionNode_ = newVersionNode;}
+
 inline void DiffComparisonPair::setNewVersionObjectPath(Model::Text* newVersionObjectPath)
 {newVersionObjectPath_ = newVersionObjectPath;}
 inline void DiffComparisonPair::setOldVersionObjectPath(Model::Text* oldVersionObjectPath)
 {oldVersionObjectPath_ = oldVersionObjectPath;}
 
+inline void DiffComparisonPair::setComponentType(Model::Text* componentType) {componentType_ = componentType; }
 
 inline Model::Node* DiffComparisonPair::newVersionNode() {return newVersionNode_;}
 inline Model::Node* DiffComparisonPair::oldVersionNode() {return oldVersionNode_;}
+
 inline Model::Text* DiffComparisonPair::newVersionObjectPath() {return newVersionObjectPath_;}
 inline Model::Text* DiffComparisonPair::oldVersionObjectPath() {return oldVersionObjectPath_;}
 
+inline Model::Text* DiffComparisonPair::componentType() {return componentType_;}
```

Comment:
This will change in the next iteration. It will then be smarter (using if applicable CompositeNodes meta information).